src/main/java/org/alliancegenome/curation_api/services/associations/alleleAssociations/AlleleGeneAssociationService.java

```diff
@@ -80,20 +91,40 @@ public List<Long> getAlleleGeneAssociationsByDataProvider(BackendBulkDataProvide
 		return associationIds;
 	}
 
-	public void removeAssociation(Long id, String dataProviderName, String md5sum) {
+	public AlleleGeneAssociation deprecateOrDeleteAssociation(Long id, Boolean throwApiError, String loadDescription, Boolean deprecate) {
 		AlleleGeneAssociation association = alleleGeneAssociationDAO.find(id);
-		if (association != null) {
-			association.setSubject(null);
-			Long noteId = null;
-			if (association.getRelatedNote() != null)
-				noteId = association.getRelatedNote().getId();
-			alleleGeneAssociationDAO.remove(id);
-			if (noteId != null)
-				noteDAO.remove(noteId);
-		} else {
-			log.error("Failed getting allele-gene association: " + id);
+		
+		if (association == null) {
+			String errorMessage = "Could not find AlleleGeneAssociation with id: " + id;
+			if (throwApiError) {
+				ObjectResponse<AlleleGeneAssociation> response = new ObjectResponse<>();
+				response.addErrorMessage("id", errorMessage);
+				throw new ApiErrorException(response);
+			}
+			log.error(errorMessage);
+			return null;
+		}
+		if (deprecate) {
+			if (!association.getObsolete()) {
+				association.setObsolete(true);
+				if (authenticatedPerson.getOktaEmail() != null) {
+					association.setUpdatedBy(personService.findPersonByOktaEmail(authenticatedPerson.getOktaEmail()));
+				} else {
+					association.setUpdatedBy(personService.fetchByUniqueIdOrCreate(loadDescription));
```

Comment:
Are we creating a bunch of "persons" based on loadDescriptions? I don't think this is a good idea as in the person table is not really "persons" anymore?

Reply:
Yeah, not sure what the best solution is without a model change though.  We've been doing the same for disease annotations as we want the "UpdatedBy" to indicate why the annotation was changed to obsolete and we don't have a person associated with a bulk load.

Reply:
We also create "people" from whatever gets submitted in the updatedBy and createdBy fields in bulk submissions, which are often not really people either.